tests/lib/prepare-uc22.sh

```diff
@@ -188,19 +188,23 @@ rm -r $snapddir
 # next re-pack the kernel snap with this version of ubuntu-core-initramfs
 
 # extract the kernel snap, including extracting the initrd from the kernel.efi
-kerneldir=/tmp/kernel-workdir
-"$EXTTESTSLIB/repack-kernel" extract upstream-pc-kernel.snap $kerneldir
+kerneldir="$(mktemp --tmpdir -d kernel-workdirXXXXXXXXXX)"
+trap 'rm -rf "${kerneldir}"' EXIT
 
-# copy the skeleton from our installed ubuntu-core-initramfs into the initrd 
-# skeleton for the kernel snap
-cp -ar /usr/lib/ubuntu-core-initramfs/main/* "$kerneldir/skeleton/main"
-
-# repack the initrd into the kernel.efi
-"$EXTTESTSLIB/repack-kernel" prepare $kerneldir
+unsquashfs -f -d "${kerneldir}" upstream-pc-kernel.snap
+(
+    cd "${kerneldir}"
+    config="$(echo config-*)"
+    kernelver="${config#config-}"
+    objcopy -O binary -j .linux kernel.efi kernel.img-"${kernelver}"
+    ubuntu-core-initramfs create-initrd --kerneldir modules/"${kernelver}" --kernelver "${kernelver}" --firmwaredir firmware --output ubuntu-core-initramfs.img
+    ubuntu-core-initramfs create-efi --initrd ubuntu-core-initramfs.img --kernel kernel.img --output kernel.efi --kernelver "${kernelver}"
```

Comment:
Should it be `--kernel kernel.img-"${kernelver}"` ? given that's where the kernel image got extracted to?

Reply:
`        args.kernel = "-".join([args.kernel, args.kernelver])`

Reply:
How confusing! Especially since it's probably me who wrote that.